lib/matplotlib/axis.py

```diff
@@ -1726,14 +1727,17 @@ def set_label_position(self, position):
     def _update_label_position(self, bboxes, bboxes2):
         """
         Update the label position based on the sequence of bounding
-        boxes of all the ticklabels
+        boxes of all the ticklabels or using the axis spines if no
+        ticklabels are present
         """
         if not self._autolabelpos:
             return
         x, y = self.label.get_position()
         if self.label_position == 'bottom':
             if not len(bboxes):
-                bottom = self.axes.bbox.ymin
```

Comment:
Would it make more sense to just throw the bounding box of the spine and the axes in to the bboxes list and let the `union` call take care of this logic?